gematria/datasets/exegesis_benchmark.cc

```diff
@@ -291,10 +309,13 @@ int main(int Argc, char *Argv[]) {
           Benchmark::Latency, State, BenchmarkPhaseSelectorE::Measure,
           BenchmarkRunner::ExecutionModeE::SubProcess, 30, {}, Benchmark::Min));
 
-  if (pfm::pfmInitialize())
-    ExitOnErr(llvm::make_error<StringError>(inconvertibleErrorCode(),
-                                            "Failed to initialize libpfm"));
+  auto JsonMemoryBuffer = ExitOnErr(
+      errorOrToExpected(MemoryBuffer::getFile(AnnotatedBlocksJson, true)));
+
+  auto ParsedAnnotatedBlocks =
+      ExitOnErr(json::parse(JsonMemoryBuffer->getBuffer()));
 
+  size_t BlockIndex = 0;
   for (const auto &AnnotatedBlock : *ParsedAnnotatedBlocks.getAsArray()) {
```

Comment:
This would be easier with a regular for loop with an index since we have the index anyway. With the indexed for loop, we'd at least be safe from `continue`.